app/views/layouts/_header.html.erb

```diff
@@ -29,13 +29,13 @@
         <% if @current_user && (@current_user.role.eql? 'admin') %>
           <li>
             <%= form_tag admin_inventories_path, method: :get do %>
-              <%= select_tag 'branch', options_for_select(Branch.all.pluck('name'), session[:current_location]), onchange: 'this.form.submit()', class: 'location' %>
+              <%= select_tag 'branch', options_for_select(Branch.all.pluck('name'), cookies[:current_location]), onchange: 'this.form.submit()', class: 'location' %>
```

Comment:
1. No need to use :onchange here. Load the selector in Js and listen event there itself.
2. No need to use Model#all. We can directly use pluck. Also, use symbol.